packages/unmock-core/src/nock.ts

```diff
@@ -198,74 +188,103 @@ type UpdateCallback = ({
 type Primitives = string | number | boolean;
 type InputToPoet = { [k: string]: any } | Primitives | Primitives[];
 
-export class DynamicServiceSpec {
+// How the fluent dynamic service API looks like (e.g. specifies `get(endpoint: string) => DynamicServiceSpec`)
+type FluentDynamicService = {
+  [k in HTTPMethod]: (endpoint: string) => DynamicServiceSpec;
+};
+
+// How the actual dynamic service spec looks like (e.g. `reply(statusCode: number, data: InputToPoet): ...`)
+//                                                      `replyWithFile(....)`
+interface IDynamicServiceSpec {
+  /**
+   * Sets the reply schema for the previous base URL, endpoint and HTTP method.
+   * @param statusCode
+   * @param data
+   */
+  reply(
+    statusCode: CodeAsInt | "default",
+    data?: InputToPoet | InputToPoet[],
+  ): FluentDynamicService & IDynamicServiceSpec;
+  reply(
+    data: InputToPoet | InputToPoet[],
+  ): FluentDynamicService & IDynamicServiceSpec;
+}
+
+export class DynamicServiceSpec implements IDynamicServiceSpec {
   private data: Schema = {};
 
   // Default status code passed in constructor
   constructor(
     private updater: UpdateCallback,
-    private statusCode: number = 200,
+    private statusCode: CodeAsInt | "default" = 200,
     private baseUrl: string,
+    private serviceStore: ServiceStore,
     private name?: string,
   ) {}
 
-  // TODO: Should this allow fluency for consecutive .get, .post, etc on the same service?
-  public reply(
-    statusCode: number,
-    data?: InputToPoet | InputToPoet[],
-  ): FluentDynamicService;
-  public reply(data: InputToPoet | InputToPoet[]): FluentDynamicService;
   public reply(
-    maybeStatusCode: number | InputToPoet | InputToPoet[],
+    maybeStatusCode: CodeAsInt | "default" | InputToPoet | InputToPoet[],
     maybeData?: InputToPoet | InputToPoet[],
-  ): FluentDynamicService {
+  ): FluentDynamicService & IDynamicServiceSpec {
     if (maybeData !== undefined) {
-      this.data = JSONSchemify(maybeData) as Schema;
-      this.statusCode = maybeStatusCode as number;
+      this.data = JSONSchemify(maybeData) as Schema; // TODO should this be some JSSTX?
```

Comment:
Is the code a good place for TODOs like this, are we planning to change to JSSTX at some point? 🤔

Reply:
Good question. `JSSTX` elements are all over the place (Mike's own `JSSTAnything, JSSTEmpty`, etc). I'm a bit tempted for us to get rid of OAS at this stage, and `Schema` comes from `loas`. Just wanted to put it there for later review.

Reply:
Ok, I'd maybe just keep those in issues then instead of adding bunch of TODOs in the code. But it doesn't matter.